src/storage/src/source/postgres/replication.rs

```diff
@@ -404,14 +411,38 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
                                     continue;
                                 }
 
-                                let data = (oid, event);
+                                let event = match event {
+                                    Ok(cols) => {
+                                        row_temp.clear();
+                                        for c in cols {
+                                            let c = c.map(|c| {
+                                                let mut col_vec =
+                                                    col_temp.pop().unwrap_or_default();
+                                                col_vec.clear();
+                                                col_vec.extend_from_slice(&c);
+                                                col_vec
+                                            });
+                                            row_temp.push(c);
+                                        }
+                                        Ok(std::mem::take(&mut row_temp))
```

Comment:
this works because Vec::default() doesn't allocate, right?

Reply:
That's right!